### ISSUE TITLE
How to successfully make request

### ISSUE THREAD

**Issue by jim256** [ghardin1314/autotrader_scrape#1]
I absolutely love what you're building here!

I'm having trouble actually making the request and was wondering if you had any suggestions? I can enter the URL in my browser just fine and get the JSON payload back, but when I try to request it in a very simplified format like the following snipped out of get_results.py _get_page:

```
make_code = 'AUDI'
model_code = 'S4'
firstRecord = 1


url_base = 'https://www.autotrader.com'
path = '/rest/searchresults/base'
url = url_base + path
session = requests.Session()
payload = {'makeCodeList': make_code,
            'searchRadius': '0',
            'listingTypes': 'USED',
            'modelCodeList': model_code,
            'zip': '90250',
            'startYear': '2010',
            'sortBy': 'distanceASC',
            'numRecords': '100',
            'firstRecord': firstRecord,
            }
headers= {'User-Agent': """Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"""}

print(f'Requesting {url}')

response = session.get(url, headers=headers, params=payload)

print(f'response: {response}')
```
, it will either time out, give a 504, or return me error in the HTML saying the page is unavailable.

Are you using some sort of proxy, cookies, other headers or something?

**Comment by ghardin1314** [comment]
Hey Jim, Im glad you like the project. Sorry its not well documented, I didnt think anyone would ever see it. I think maybe the problem is that your firstRecord is an int and it needs to be transformed into a string for the json request. If that doesnt work let me know.

Edit: I added 'import requests' to the top of your code and ran it just fine so thats not the issue. I am not using any special proxy or anything, but you might need one depending on your location. Also the api will shut you down if you make too many requests too quickly. Not sure thats an issue, but just for warning

**Comment by jim256** [comment]
hey @ghardin1314 , so sorry I missed your comment above!
I appreciate your getting back to me and trying my snippet. What has your experience been in terms of how many requests you can make before the API shuts you down? I'd like to respect their limitations.

**Comment by jim256** [comment]
Also, I'd love to get a better sense of everything this repo does. It looks really cool. What types of analysis does it do? Would love to just get an overview of how it's mean to be used / it's features :)

**Comment by ghardin1314** [comment]
@jim256 I'm not sure exactly what there limits are now. I had to put a `time.sleep(10)` when pulling the list of all the trims in cars_thread_db.py to keep it from shutting down. Other than that I haven't had any issues using it. 

I'm actually building this into a web app right now so I looped through all the makes, models, and trims and pull the first 1000 listings of each one using the code in get_results.py to store in a database. I'm going to use that database so I don't have to ping them every time someone visits my site. 

Basically the second half of get_results.py takes the price, mileage, and year data and creates a cubic surface to approximate results. Then finds minimum curvature of the surface inside of the Year-Miles domain range. This is the place where depreciation is decelerating the most and therefore the optimal place to buy a car.

My web app repository is going to be [https://github.com/ghardin1314/GRHrest](here) if you want to follow. Still working to get all the framework up but it should be much cleaner when its done.
